server/src/repositories/patient_repository.ts

```diff
@@ -25,4 +25,20 @@ export class PatientRepository {
         await client.query("COMMIT;");
         client.release();
     }
+
+    async assignDoctor(patientId: number, doctorId: number): Promise<void> {
+        const client = await this.pool.connect();
+
+        await client.query("BEGIN;");
```

Comment:
You don't have to wrap a single query in a transaction (begin and commit), it will slow it down for no benefit because its a single unit of work with a single part that doesn't have a rollback.